pkg/proxy/proxy/transparent.go

```diff
@@ -87,31 +87,40 @@ func (m PolicyMode) IsValid() bool {
 
 // TransparentProxyService transparently proxies HTTP and HTTPS traffic.
 type TransparentProxyService struct {
-	Proxy      *goproxy.ProxyHttpServer
-	Ca         *tls.Certificate
-	NetworkLog *netlog.NetworkActivityLog
-	Policy     *policy.Policy
-	Mode       PolicyMode
+	Proxy  *goproxy.ProxyHttpServer
+	Ca     *tls.Certificate
+	Policy *policy.Policy
+	Mode   PolicyMode
 
-	mx *sync.Mutex
+	mx         *sync.Mutex
+	networkLog *netlog.NetworkActivityLog
+}
+
+// TransparentProxyServiceOpts defines the optional parameters for creating a TransparentProxyService.
+type TransparentProxyServiceOpts struct {
+	SkipLogInit bool
 }
 
 // NewTransparentProxyService creates a new TransparentProxyService.
-func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate, mode PolicyMode, pl *policy.Policy) TransparentProxyService {
+func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate, mode PolicyMode, pl *policy.Policy, opts TransparentProxyServiceOpts) TransparentProxyService {
```

Comment:
i was also imagining the policy configuration moving to the Opt struct but we could also consider that a follow-on.

Reply:
I moved the policy inside the opt